Newsletters: add 402 (2026-04-24) (#2725)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2026-04-24-newsletter.md

name: 2026-04-24-newsletter
slug: 2026-04-24-newsletter
type: newsletter
layout: newsletter
lang: en
---
FIXME:bitschmidty

## News

- **Hornet Update: A declarative executable specification of consensus rules**:
  Toby Sharp [posted][topic hornet update] to Delving Bitcoin and the mailing
  list an update on the Hornet node project. Sharp had
  [previously][topic hornet] described a new node implemntation Hornet that
  reduced initial block download times from 167 minutes down to 15 minutes. In
  this update, he reports completing a declarative specification of non-script
  block validation rules, consisting of 34 semantic invariants composed using a
  simple algebra.

  Sharp also outlines future work, including extending the specification to
  script validation, and discusses potential comparisons with other
  implementations such as libbitcoin in response to feedback from Eric Voskuil.

- **Onion message jamming in the lightning network**: Erick Cestari [posted][onion del] to
  Delving Bitcoin about the [onion message][topic onion messages] jamming problem affecting
  the lightning network. BOLT4 acknowledges onion messages to be unreliable, recommending

>   implementation to apply rate limiting techniques in the different implementations.

Reviewer: ```suggestion
  to apply rate limiting techniques in different implementations.
```

    as a solution for channel jamming, but can be easily extended. Nodes would advertise a per-message
    flat fee, to be included in the onion payload and deducted at each hop. In case the fee is not
    paid, the message would be just dropped by the node. This method presents some limitations, such
    as being able to only forward messages to channel peers and increased p2p overhead.

  - **Hop limit and proof of stake based on channel balances**: This technique was [proposed][mitig2 onion]
    by Bashiri and Khabbazian at the University of Alberta and has two different components:
    - Leashing hop count: Either setting a hard-cap on the maximum number of hops that a message could do
      (e.g. 3 hops), or having the sender solve a proof of work puzzle, whose difficulty increases
      exponentially with the number of hops.
    - Proof of stake forwarding rule: Each node sets per-peer rate limits according to the peer's aggregate
      channel balance, granting well-funded nodes more forwarding power.

    The trade-offs of this approach are related to centralization pressure, due to larger nodes being at

>     an advantage, while the 3-hops hard-cap could reduce anonimity set.

Reviewer: ```suggestion
    an advantage, while the 3-hops hard-cap could reduce anonymity set.
```

    similar in scope as the upfront fees one, but adds per-session state and settles through
    [AMP payments][topic amp]. A sender would first send the AMP payment, dropping fees
    at each intermediate step and delivering an id for the session. The sender would then include the
    id in the onion message. Known limitation of the approach are related to the ability to only forward
    messages to channel peers and the possibility to link all the messages belonging to the same session.

  - **Backpropagation-based rate limiting**: This approach, [proposed][mitig4 onion] by Bastien Teinturier,
    uses a backpressure mechanism that is statistically able to trace back spam to its source.
    When the per-peer rate limiting is hit, the node send a drop message back to the sender, which in turn
    relays it to the last peer that forwarded the original message halving its rate limit. While the correct
    sender is statistically identified, the wrong peer could be penalized. Moreover, an attacker could fake
    the drop message, lowering rate limits of honest nodes.

  Finally, Cestari invited LN developers to join the discussion, stating that a window is still available to

>   mitigate the issue before prolonged DDos attacks hit the network, as [happened to Tor][tor issue]

Reviewer: ```suggestion
  mitigate the issue before prolonged DDoS attacks hit the network, as [happened to Tor][tor issue]
```

  mempool][topic cluster mempool] design, a new `-privatebroadcast` option for
  `sendrawtransaction` (see [Newsletter #388][news388 private]), `asmap` data
  optionally embedded into the binary for [eclipse attack][topic eclipse
  attacks] protection, and an increase in the default `-dbcache` to 1024 MiB on
  systems with at least 4096 MiB of RAM, among many other updates.

- [Core Lightning 26.04][] is a major release of this popular LN node
  implementation. It enables [splicing][topic splicing] by default, adds new
  `splicein` and `spliceout` commands including a `cross-splice` mode that
  targets a second channel as the destination of a splice-out, redesigns
  `bkpr-report` for income summaries, adds parallel pathfinding and multiple bug
  fixes in `askrene`, adds a `fronting_nodes` option on the `offer` RPC and a
  `payment-fronting-node` config, and removes support for the legacy onion
  format. See the [release notes][notes cln] for additional details.

- [LND 0.21.0-beta.rc1][] is the first release candidate for the next major
  version of this popular LN node. Users running nodes with `--the

>   db.use-native-sql` flag against an SQLite or Postgres backend should note that

Reviewer: ```suggestion
  version of this popular LN node. Users running nodes with the
  `--db.use-native-sql` flag against an SQLite or PostgreSQL backend should note that
```

---
title: 'Bitcoin Optech Newsletter #402'
permalink: /en/newsletters/2026/04/24/
name: 2026-04-24-newsletter
slug: 2026-04-24-newsletter
type: newsletter
layout: newsletter
lang: en
---
FIXME:bitschmidty

## News

- **Hornet Update: A declarative executable specification of consensus rules**:
  Toby Sharp [posted][topic hornet update] to Delving Bitcoin and the mailing
  list an update on the Hornet node project. Sharp had

>   [previously][topic hornet] described a new node implemntation Hornet that

Reviewer 2: ```suggestion
  [previously][topic hornet] described a new node implementation Hornet that
```

  this update, he reports completing a declarative specification of non-script
  block validation rules, consisting of 34 semantic invariants composed using a
  simple algebra.

  Sharp also outlines future work, including extending the specification to
  script validation, and discusses potential comparisons with other
  implementations such as libbitcoin in response to feedback from Eric Voskuil.

- **Onion message jamming in the lightning network**: Erick Cestari [posted][onion del] to
  Delving Bitcoin about the [onion message][topic onion messages] jamming problem affecting
  the lightning network. BOLT4 acknowledges onion messages to be unreliable, recommending
  implementation to apply rate limiting techniques in the different implementations.
  According to Cestari, these techniques are what makes message jamming possible. Attackers
  may spin up malicious nodes and flood the network with spam messages triggering rate limits
  of peers, forcing them to drop legitimate messages. Moreover, BOLT4 does not enforce a
  maximum message length, allowing attackers to maximize the reach of a single message.

>   Cestari reveiewed several mitigations to onion message jamming and provides a

Reviewer 2: ```suggestion
  Cestari reviewed several mitigations to onion message jamming and provides a
```

  before it reaches the onion handler. This hardens LND's recently added onion
  message forwarding support (see [Newsletter #396][news396 lnd onion]) against
  high-volume abuse from fast peers. The per-peer and global split mirrors LND's
  earlier gossip bandwidth limits (see [Newsletter #370][news370 lnd gossip]).

- [LND #10754][] stops forwarding an [onion message][topic onion messages] when
  the chosen next hop is the same peer that delivered it, avoiding an immediate
  bounce on the same connection.

- [BIPs #1895][] publishes [BIP361][], an abstract proposal for
  [post-quantum][topic quantum resistance] migration and legacy signature
  sunset. Assuming a separate post-quantum (PQ) signature scheme is standardized
  and deployed, it outlines a phased migration away from ECDSA/
  [Schnorr][topic schnorr signatures] signature schemes. The current version of the proposal is
  divided into two phases. Phase A prohibits sending funds to quantum-vulnerable
  addresses, thereby accelerating the adoption of PQ address types. Phase B
  restricts ECDSA/Schnorr spending by imposing a quantum-safe rescue protocol on
  them to prevent the theft from quantum-vulnerable UTXOs.

> - [BIPs #2142][] updates [BIP352][], the

Reviewer 2: Did we want to go ahead with the reordering of the notable code and documentation changes section this time already?

Reviewer 3: right, on it
